addon/model/model-range.ts

```diff
@@ -311,6 +311,138 @@ export default class ModelRange {
     }
   }
 
+  getTextContent(): string {
+    return this.textContentHelper(false).textContent;
+  }
+
+  getTextContentWithMapping(): {
+    textContent: string;
+    indexToPos: (textIndex: number) => ModelPosition;
+  } {
+    return this.textContentHelper(true);
+  }
+
+  private textContentHelper(calculateMapping: boolean): {
+    textContent: string;
+    indexToPos: (textIndex: number) => ModelPosition;
+  } {
+    let textContent = '';
+
+    // a sparse map of character indices of the resultstring to paths
+    // it looks something like this:
+    // [[4, [0, 0]], [18, [1,3,4]]]
+    // the first number is the limitnumber
+    // meaning all indices up to 4 correspond to textnode with path [0,0],
+    // all indices between 4 and 18 correspond to textnode with path [1,3,4], etc
+    // to get the final path of the index, add the difference between the index
+    // and the last encountered limit to the last element of the path
+    const mapping: [number, number[]][] = [];
+
+    // get all textnodes in range
+    const walker = GenTreeWalker.fromRange({
+      range: this,
+      descend: true,
+    });
+
+    // calculate difference between start of range and start of the first textnode
+    const startOffset = this.start.isInsideText()
+      ? this.start.parentOffset - this.start.nodeAfter()!.getOffset()
+      : 0;
+
+    let currentIndex = 0;
+    // build the resultstring
+    for (const node of walker.nodes()) {
+      if (ModelNode.isModelText(node)) {
+        // keep a sparse mapping of character indices in the resulting string to paths
+        const sanitizedContent = node.content.replace(/\u200B/g, '');
```

Comment:
I'd prefer the use of constants for things like the invisible space, makes it more clear what is is

Reply:
yeah but you cant do that in a regex

Reply:
```
const myRegex = new Regex(`${MYCONSTANT}`,'g');
node.content.replace(myRegex, '');
```
should work, no?

Reply:
yes indeed, good idea!